Assets/_Laz/Scripts/UserInterface/DebugBehaviour/DebugLazMovement/DebugMovementParametersEditor.cs

```diff
@@ -0,0 +1,312 @@
+                GUILayout.FlexibleSpace();
+            }
+            
+        }
+        
+        //Can only create GUIStyles on GUICalls
+        //So Create GUIStyles on first call
+        private void CreateGUIStyles()
+        {
+            if (_headerGUIStyle != GUIStyle.none && _labelGUIStyle != GUIStyle.none &&
+                _valueGUIStyle != GUIStyle.none && _fileNameStyle != GUIStyle.none)
+            {
+                return;
+            }
+            
+            _headerGUIStyle = new GUIStyle(GUI.skin.label)
+            {
+                fontStyle = FontStyle.Bold,
+                alignment = TextAnchor.MiddleCenter,
+                normal = new GUIStyleState
+                {
+                    textColor = Color.black
+                },
+                stretchWidth = true,
+                stretchHeight = true,
+            };
+
+            _labelGUIStyle = new GUIStyle(GUI.skin.label)
+            {
+                fontStyle = FontStyle.Bold,
+                alignment = TextAnchor.MiddleCenter,
+                normal =  new GUIStyleState
+                {
+                    textColor = Color.white
+                }
+            };
+
+            _valueGUIStyle = new GUIStyle()
+            {
+                stretchWidth = false,
+                stretchHeight = true,
+                normal = new GUIStyleState
+                {
+                    textColor = Color.white
+                }
+            };
+
+            RectOffset fileNamePadding = GUI.skin.textField.padding;
+            fileNamePadding.top = 4;
+            
+            _fileNameStyle = new GUIStyle(GUI.skin.textField)
+            {
+                padding = fileNamePadding
+            };
+        }
+
+        private void CreateGUIParameters()
+        {
+            if (_rectTransform == null)
+            {
+                _rectTransform = GetComponent<RectTransform>();
+            }
+
+            _rect = new Rect(_rectTransform.anchorMin.x * Screen.width, _rectTransform.anchorMin.y * Screen.height,
+                _rectTransform.rect.width, _rectTransform.rect.height);
+
+            _headerHeight = _rect.height * HEADER_HEIGHT_MULTIPLIER;
+            _headerPadding = _rect.width * HEADER_PADDING_MULTIPLIER;
+
+            _bodyHeight = _rect.height * BODY_HEIGHT_MULTIPLIER;
+            
+            _footerHeight = _rect.height * FOOTER_HEIGHT_MULTIPLIER;
+        }
+
+        #endregion
+
+        private void ResetContent()
+        {
+            _movementProperty = ScriptableObject.CreateInstance<LazMovementPropertyScriptableObject>();
+            _fileName = string.Empty;
+        }
+
+        private void SaveMovementProperty()
+        {
+            if (string.IsNullOrEmpty(_fileName))
+            {
+                Debug.LogWarning("Please fill in file name before attempting to Save");
+                return;
+            }
+
+            Writer.WriteToFile(_fileName, _movementProperty);
+        }
```

Comment:
This seems a little unintuitive in-editor to me. If the user tries to save in the editor, why do we need to go through a json file? Would it be possible to wrap this in a build conditional and directly create asset files in editor?